ipam/allocator.go

```diff
@@ -696,8 +700,12 @@ func (alloc *Allocator) actorLoop(actionChan <-chan func(), stopChan <-chan stru
 			// Retry things in case messages got lost between here and recipients
 			if alloc.awaitingConsensus {
 				alloc.propose()
-			} else {
-				alloc.tryPendingOps()
+			} else if alloc.havePendingOps() {
+				if alloc.ring.Empty() {
+					alloc.establishRing()
+				} else {
+					alloc.tryPendingOps()
+				}
```

Comment:
> surely we'd want to call tryPendingOps() immediately after establishing the ring

`ringUpdated()` calls `tryPendingOps()`, so that isn't necessary.  Nonetheless your version is one line shorter, and perhaps clearer, so I changed it.